test/utils.cpp

```diff
@@ -119,6 +119,17 @@ TEST_CASE("Tokenizers behave correctly", "[utils]") {
 	}
 }
 
+TEST_CASE("utils::consolidate_whitespace replaces multiple consecutinve whitespace with a single space", "[utils]") {
+	REQUIRE(utils::consolidate_whitespace("LoremIpsum") == "LoremIpsum");
+	REQUIRE(utils::consolidate_whitespace("Lorem Ipsum") == "Lorem Ipsum");
+	REQUIRE(utils::consolidate_whitespace("    Lorem \t\tIpsum \t ") == " Lorem Ipsum ");
+	REQUIRE(utils::consolidate_whitespace("    Lorem \r\n\r\n\tIpsum") == " Lorem Ipsum");
+
+	REQUIRE(utils::consolidate_whitespace("") == " ");
```

Comment:
We do rely on it, according to my tests, empty lines don't get rendered if the string is empty.